tests/tracer/test_logging.py

```diff
@@ -0,0 +1,284 @@
+
+assert custom_logger.parent.name == 'root'
+assert custom_logger.level == logging.WARN
+
+ddtrace_logger = logging.getLogger('ddtrace')
+assert ddtrace_logger.level == logging.DEBUG
+assert ddtrace_logger.handlers[0].__class__.__name__ == 'RotatingFileHandler'
+assert ddtrace_logger.handlers[0].maxBytes == 15 << 20
+"""
+
+    out, err, status, pid = run_python_code_in_subprocess(code, env=env)
+    assert status == 0, err
+    assert err == b""
+    assert out == b""
+    with open(log_file) as file:
+        content = file.read()
+        assert len(content) > 0
+
+
+def test_child_logger_inherits_settings_patch_all(run_python_code_in_subprocess, tmpdir):
+    """
+    When setting up a child logger under ddtrace with debug mode and log path enabled,
+        the child logger inherits from ddtrace automatically.
+    """
+    env = os.environ.copy()
+    env["DD_TRACE_DEBUG"] = "true"
+    log_file = tmpdir + "/testlog.log"
+    env["DD_TRACE_LOG_FILE"] = log_file
+    code = """
+import logging
+import ddtrace
+
+ddtrace.patch_all()
+
+ddtrace_logger = logging.getLogger('ddtrace')
+assert ddtrace_logger.level == logging.DEBUG
+assert ddtrace_logger.handlers[0].__class__.__name__ == 'RotatingFileHandler'
+assert ddtrace_logger.handlers[0].maxBytes == 15 << 20
+
+child_logger = logging.getLogger('ddtrace.child')
+assert child_logger.parent.name == 'ddtrace'
+assert child_logger.getEffectiveLevel() == logging.DEBUG
+assert child_logger.handlers == []
+"""
+
+    out, err, status, pid = run_python_code_in_subprocess(code, env=env)
+    assert status == 0, err
+    assert err == b""
+    assert out == b""
+    with open(log_file) as file:
+        content = file.read()
+        assert len(content) > 0
+
+
+def test_debug_logs_go_to_stderr_patch_all(run_python_code_in_subprocess):
+    """
+    When setting up the default logger in debug mode,
+        it automatically logs to stderr.
+    """
+    env = os.environ.copy()
+    env["DD_TRACE_DEBUG"] = "true"
+    code = """
+import logging
+import ddtrace
+
+ddtrace.patch_all()
+
+ddtrace_logger = logging.getLogger('ddtrace')
+assert ddtrace_logger.level == logging.DEBUG
+assert ddtrace_logger.handlers[0].__class__.__name__ == 'StreamHandler'
+"""
+
+    out, err, status, pid = run_python_code_in_subprocess(code, env=env)
+    assert status == 0, err
+    assert err != b""
+    assert out == b""
+
+
+def test_warn_logs_go_to_stderr_patch_all(run_python_code_in_subprocess):
+    """
+    When setting up the default logger,
+        it automatically logs to stderr at WARN log level.
+    """
+    code = """
+import logging
+import ddtrace
+
+ddtrace.patch_all()
+
+ddtrace_logger = logging.getLogger('ddtrace')
+assert ddtrace_logger.level == logging.WARN
+assert ddtrace_logger.handlers[0].__class__.__name__ == 'StreamHandler'
+
+ddtrace_logger.warning('warning log')
+"""
+
+    out, err, status, pid = run_python_code_in_subprocess(code)
+    assert status == 0, err
+    assert err != b""
```

Comment:
🤔 I wonder if we should do something like:

```python
assert b"warning log" in err
```

and maybe the same for the other test as well, what do you think?

I am less a fan of it if we are using `patch_all()` to simulate logging, but for a basic example like this logging a single log... we can probably even just do a regex to match the specific log format string too ?

Reply:
I added `assert b"warning log" in err` to the newer tests, so I'm going to mark this as resolved.